File: esp32/components/graphics_lib/graphics/ui/text/text.cpp
#include "text.h"

#include <utility>

UIText::UIText(std::string text) : text(std::move(text))
{

}

std::string UIText::render(const bool selected)
{
    return (selected ? ">" : "") + text;
}

void UIText::setText(std::string newText)
{
    text = std::move(newText);
}

File: esp32_rust/components/main_rust/src/graphics/ui/text.rs
use crate::graphics::ui::element::{UIElement, UIElementState};

pub struct UIText {
    text: String
}

impl UIText {
    pub fn new(value: String) -> UIText {
        UIText { text: value }
    }
}

impl UIElement for UIText {
    fn render(&self, state: UIElementState) -> String {
        let prefix = match state {
            UIElementState::Selected => ">",
            UIElementState::SelectorPress => "\\",
            UIElementState::None => ""
        };
        format!("{}{}", prefix, self.text)
    }
}
